tests/test_runlib.py

```diff
@@ -726,6 +814,40 @@ def test_normalize_line_endings(self):
       for path in paths:
         os.remove(path)
 
+  def test_wrong_metadata_directory(self):
+    """Test record stop with wrong metadata directory"""
+    if sys.version_info < (3, 0):
+      with self.assertRaises(IOError):
+        in_toto_record_start(self.step_name, [], self.key)
+        in_toto_record_stop(self.step_name, [], self.key,
+            metadata_directory='this-is-not-a-dir')
+    else:
+      with self.assertRaises(FileNotFoundError):
+        in_toto_record_start(self.step_name, [], self.key)
+        in_toto_record_stop(self.step_name, [], self.key,
+            metadata_directory='this-is-not-a-dir')
```

Comment:
I suggest to just put the things that differ in the if/else clause, e.g.:
```python
expected_error = IOError if  sys.version_info < (3, 0) else FileNotFoundError
```
Same applies to `test_read_only_metadata_directory`.